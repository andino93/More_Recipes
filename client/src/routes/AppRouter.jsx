import React from 'react';
import { Router, Route, Switch } from 'react-router-dom';
import history from '../utils/history';
import HomePageComponent from '../components/Homepage.jsx';
import UserRecipeComponent from '../components/UserRecipe.jsx';
import RecipeDetailComponent from '../components/RecipeDetail.jsx';
import ProfileComponent from '../components/Profile.jsx';
import FavoriteComponent from '../components/Favorites.jsx';
import AddRecipeComponent from '../components/AddRecipe.jsx';
import EditRecipeComponent from '../components/EditRecipe.jsx';
import NotFound from '../components/NotFound.jsx';
import PrivateRoute from "./PrivateRoute.jsx";

const AppRouter = () => (
  <Router history={history}>
    <div>
      <Switch>
        <Route exact path="/" component={HomePageComponent} />
        <PrivateRoute exact path="/profile" component={ProfileComponent} />
        <PrivateRoute exact path="/favorite" component={FavoriteComponent} />
        <PrivateRoute exact path="/user-recipes" component={UserRecipeComponent} />
        <PrivateRoute exact path="/addRecipe" component={AddRecipeComponent} />
        <PrivateRoute exact path="/recipes/:recipeId" component={RecipeDetailComponent} />
        <PrivateRoute exact path="/recipes/:recipeId/edit" component={EditRecipeComponent} />
        <PrivateRoute path="*" component={NotFound}/>
      </Switch>
    </div>
  </Router>
);

export default AppRouter;
