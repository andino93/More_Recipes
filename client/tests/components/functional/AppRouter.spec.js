import React from 'react';
import expect from 'expect';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import localStorageMock from '../../__mocks__/localStorageMock';
import AppRouter from '../../../src/routes/AppRouter.jsx';
import PrivateRoute from '../../../src/routes/PrivateRoute.jsx';
import mockData from "../../__mocks__/mockData";

configure({ adapter: new Adapter() });

// noinspection JSAnnotator
window.localStorage = localStorageMock;
window.localStorage.token = mockData.token;

describe('<AppRouter>', () => {
  it('should render AppRouter', () => {
    const wrapRouter = shallow(<AppRouter />);
    expect(wrapRouter).toMatchSnapshot();
  });
  it('should render PrivateRoute', () => {
    const wrapRouter = shallow(<PrivateRoute />);
  });
});
