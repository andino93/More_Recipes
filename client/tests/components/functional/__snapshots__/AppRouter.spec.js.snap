// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter> should render AppRouter 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/profile"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/my-favorite"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/favorites"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/user-recipes"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/addRecipe"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/recipes/:recipeId"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/recipes/:recipeId/edit"
      />
      <Route
        component={[Function]}
        path="*"
      />
    </Switch>
  </div>
</Router>
`;
